x/accounts/internal/implementation/protoaccount.go

```diff
@@ -33,27 +33,21 @@ func RegisterInitHandler[
 // RegisterExecuteHandler registers an execution handler for a smart account that uses protobuf.
 func RegisterExecuteHandler[
 	Req any, ProtoReq ProtoMsgG[Req], Resp any, ProtoResp ProtoMsgG[Resp],
-](router *ExecuteBuilder, handler func(ctx context.Context, req ProtoReq) (ProtoResp, error),
+](router ProtoMsgHandlerRegistry, handler func(ctx context.Context, req ProtoReq) (ProtoResp, error),
 ) {
 	reqName := MessageName(ProtoReq(new(Req)))
-	// check if not registered already
-	if _, ok := router.handlers[reqName]; ok {
-		router.err = fmt.Errorf("handler already registered for message %s", reqName)
-		return
-	}
-
-	router.handlers[reqName] = func(ctx context.Context, executeRequest transaction.Msg) (executeResponse transaction.Msg, err error) {
+	fn := func(ctx context.Context, executeRequest transaction.Msg) (executeResponse transaction.Msg, err error) {
```

Comment:
the error is returned in this helper.  the registry takes care of the duplicate prevention and internal state